[Bug]: psqlODBC driver build failure in github action

### What happened?

ODBC Tests workflow [uses psqlODBC 12](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/8c5ca4839e0d7abf78e8952d9b08a6ea297195c3/.github/workflows/odbc-tests.yml#L61) to run the test suite. It is currently failing on github actions like this ([run example](https://github.com/babelfish-for-postgresql/babelfish_extensions/actions/runs/6277810842/job/17050299156?pr=1861)):
```
make[1]: Entering directory '/home/runner/psqlodbc-12.01.0000'
/bin/bash ./libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I.   -I/usr/local/include -I/usr/include/postgresql -I/usr/include/postgresql/internal -DUNICODE_SUPPORT  -g -O2 -Wall -MT psqlodbcw_la-info.lo -MD -MP -MF .deps/psqlodbcw_la-info.Tpo -c -o psqlodbcw_la-info.lo `test -f 'info.c' || echo './'`info.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I/usr/local/include -I/usr/include/postgresql -I/usr/include/postgresql/internal -DUNICODE_SUPPORT -g -O2 -Wall -MT psqlodbcw_la-info.lo -MD -MP -MF .deps/psqlodbcw_la-info.Tpo -c info.c  -fPIC -DPIC -o .libs/psqlodbcw_la-info.o
In file included from connection.h:14,
                 from info.c:34:
/usr/include/postgresql/internal/pqexpbuffer.h:168:8: error: unknown type name 'bool'
  168 | extern bool appendPQExpBufferVA(PQExpBuffer str, const char *fmt, va_list args) pg_attribute_printf(2, 0);
      |        ^~~~
make[1]: *** [Makefile:657: psqlodbcw_la-info.lo] Error 1
make: *** [Makefile:479: all] Error 2
make[1]: Leaving directory '/home/runner/psqlodbc-12.01.0000'
Error: Process completed with exit code 2.
```

This is happening as a result of [this commit in upstream Postgres](https://github.com/postgres/postgres/commit/0873b2d354b9c73a69067c0afb92e35994adc47e). This commit is not present in latest `postgresql_modified_for_babelfish`, but psqlODBC is built not with `postgresql_modified_for_babelfish`, but with `libpq-dev`. [Ubuntu-20.04 github actions image](https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2004-Readme.md) comes with `libpq-dev` version `16.0-1.pgdg20.04+1` pre-installed (I assume it is coming from [this repo](https://apt.postgresql.org/pub/repos/apt/dists/focal-pgdg/)).

This problem was fixed in  psqlODBC 16 ([discussion](https://www.postgresql.org/message-id/ZPUatGNtOSe/kfDX%40paquier.xyz), [commit](https://git.postgresql.org/gitweb/?p=psqlodbc.git;a=commitdiff;h=4da3826ae37b810b3edc570e8f84f6895d6df80a;hp=d24ba430ec29006ed71f9bbe2676844f79839de8)). I assume it makes sense to use psqlODBC 16 in github action runs, will create a PR for this.

### Version

BABEL_3_X_DEV (Default)

### Extension

None

### Which flavor of Linux are you using when you see the bug?

Ubuntu (Default)

### Relevant log output

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct.

--- a/.github/workflows/code-coverage.yml
+++ b/.github/workflows/code-coverage.yml
@@ -168,14 +168,14 @@ jobs:
         if: always() && steps.install-extensions.outcome == 'success' && steps.install-unix-odbc-driver.outcome=='success'
         run: |
           cd ~
-          wget https://ftp.postgresql.org/pub/odbc/versions/src/psqlodbc-12.01.0000.tar.gz
-          tar -zxvf psqlodbc-12.01.0000.tar.gz
-          cd psqlodbc-12.01.0000
+          wget https://ftp.postgresql.org/pub/odbc/versions/src/psqlodbc-16.00.0000.tar.gz
+          tar -zxvf psqlodbc-16.00.0000.tar.gz
+          cd psqlodbc-16.00.0000
           ./configure
           sudo make
           sudo make install
-          echo '[ODBC_Driver_12_for_PostgreSQL]' | sudo tee -a /etc/odbcinst.ini > /dev/null
-          echo 'Description=ODBC Driver 12 for PostgreSQL Server' | sudo tee -a /etc/odbcinst.ini > /dev/null
+          echo '[ODBC_Driver_16_for_PostgreSQL]' | sudo tee -a /etc/odbcinst.ini > /dev/null
+          echo 'Description=ODBC Driver 16 for PostgreSQL Server' | sudo tee -a /etc/odbcinst.ini > /dev/null
           echo 'Driver=/usr/local/lib/psqlodbcw.so' | sudo tee -a /etc/odbcinst.ini > /dev/null
           echo 'UsageCount=1' | sudo tee -a /etc/odbcinst.ini > /dev/null
       
@@ -191,7 +191,7 @@ jobs:
             MSSQL_BABEL_DB_USER=jdbc_user \
             MSSQL_BABEL_DB_PASSWORD=12345678 \
             MSSQL_BABEL_DB_NAME=master \
-            PSQL_ODBC_DRIVER_NAME=ODBC_Driver_12_for_PostgreSQL \
+            PSQL_ODBC_DRIVER_NAME=ODBC_Driver_16_for_PostgreSQL \
             PSQL_BABEL_DB_SERVER=localhost \
             PSQL_BABEL_DB_PORT=5432 \
             PSQL_BABEL_DB_USER=jdbc_user \

--- a/.github/workflows/odbc-tests.yml
+++ b/.github/workflows/odbc-tests.yml
@@ -58,14 +58,14 @@ jobs:
         if: steps.install-extensions.outcome == 'success' && steps.install-unix-odbc-driver.outcome=='success'
         run: |
           cd ~
-          wget https://ftp.postgresql.org/pub/odbc/versions/src/psqlodbc-12.01.0000.tar.gz
-          tar -zxvf psqlodbc-12.01.0000.tar.gz
-          cd psqlodbc-12.01.0000
+          wget https://ftp.postgresql.org/pub/odbc/versions/src/psqlodbc-16.00.0000.tar.gz
+          tar -zxvf psqlodbc-16.00.0000.tar.gz
+          cd psqlodbc-16.00.0000
           ./configure
           sudo make
           sudo make install
-          echo '[ODBC_Driver_12_for_PostgreSQL]' | sudo tee -a /etc/odbcinst.ini > /dev/null
-          echo 'Description=ODBC Driver 12 for PostgreSQL Server' | sudo tee -a /etc/odbcinst.ini > /dev/null
+          echo '[ODBC_Driver_16_for_PostgreSQL]' | sudo tee -a /etc/odbcinst.ini > /dev/null
+          echo 'Description=ODBC Driver 16 for PostgreSQL Server' | sudo tee -a /etc/odbcinst.ini > /dev/null
           echo 'Driver=/usr/local/lib/psqlodbcw.so' | sudo tee -a /etc/odbcinst.ini > /dev/null
           echo 'UsageCount=1' | sudo tee -a /etc/odbcinst.ini > /dev/null
       
@@ -81,7 +81,7 @@ jobs:
             MSSQL_BABEL_DB_USER=jdbc_user \
             MSSQL_BABEL_DB_PASSWORD=12345678 \
             MSSQL_BABEL_DB_NAME=master \
-            PSQL_ODBC_DRIVER_NAME=ODBC_Driver_12_for_PostgreSQL \
+            PSQL_ODBC_DRIVER_NAME=ODBC_Driver_16_for_PostgreSQL \
             PSQL_BABEL_DB_SERVER=localhost \
             PSQL_BABEL_DB_PORT=5432 \
             PSQL_BABEL_DB_USER=jdbc_user \

